main.tf

```diff
@@ -1,4 +1,41 @@
 # ------------------------------------------------------------------------------
 # Resources
 # ------------------------------------------------------------------------------
+resource "aws_customer_gateway" "customer-gateway" {
+  bgp_asn    = "${var.customer_gateway_bgp_asn}"
+  ip_address = "${var.customer_gateway_ip}"
+  type       = "ipsec.1"
+}
 
+resource "aws_vpn_gateway" "vpn-gateway" {
+  vpc_id          = "${var.vpc_id}"
+  tags            = "${var.tags}"
+  amazon_side_asn = "${var.amazon_side_asn}"
+}
+
+data "aws_route_table" "public_subnets_route_table" {
+  route_table_id = "${var.public_subnets_route_table_id}"
+}
+
+data "aws_route_table" "private_subnets_route_tables" {
+  count          = "${var.private_subnet_count}"
+  route_table_id = "${element(var.private_subnets_route_table_ids,count.index)}"
+}
+
+resource "aws_vpn_gateway_route_propagation" "public_out" {
+  route_table_id = "${data.aws_route_table.public_subnets_route_table.route_table_id}"
+  vpn_gateway_id = "${aws_vpn_gateway.vpn-gateway.id}"
+}
+
+resource "aws_vpn_gateway_route_propagation" "private_out" {
+  count          = "${var.private_subnet_count}"
+  route_table_id = "${element(data.aws_route_table.private_subnets_route_tables.*.id,count.index)}"
```

Comment:
Not sure I understand this? You request `private_subnets_route_table_ids` from the user, and then use that list to fetch the ID of the route table? Isn't that exactly what the user has passed into the module?

Reply:
I.e., it doesn't look like you need the data sources here?